### ISSUE TITLE
No implicit dependency created for 'consul_keys' resource when utilizing a consul_keys variable

### ISSUE THREAD

**Issue by blakeneyops** [hashicorp/terraform#1852]
I am utilizing Consul to store information about an AWS VPC for use by other application stacks. An example of utilizing this information is as follows:

```
provider "consul" {
        address = "${var.registry_ip}:8500"
        datacenter = "${var.registry_dc}"
}

resource "consul_keys" "vpc" {
        key {
                name = "lan_0_id"
                path = "infrastructure/test/lan_0_id"
        }
}

module "test" {
        source = "../../../templates/aws/test"
        
        # VPC parameters
        region = "${var.region}"
        az = "${lookup(var.az,0)}"
        lan_id = "${consul_keys.vpc.var.lan_0_id}"
}
```

Plan and apply are working fine, but I am seeing an error like the following when trying to destroy the stack.

```
consul_keys.vpc: Refreshing state... (ID: consul)
module.test.aws_instance.test: Refreshing state... (ID: i-e5dff918)
consul_keys.vpc: Destroying...
consul_keys.vpc: Destruction complete
Error applying plan:

1 error(s) occurred:

* Resource 'consul_keys.vpc' does not have attribute 'var.lan_0_id' for variable 'consul_keys.vpc.var.lan_0_id'
```

From the output you can see that the 'consul_keys' resource is destroyed before the module. In checking the tfstate file, I can see that there is indeed no dependency indicated between the two resources. I would expect that use of a variable from a consul_keys resource would create an implicit dependency in this case. I tried creating the dependency myself using depends_on, but it doesn't look like it is supported for modules.

I believe this is probably related to the issue being reported in issue #491

**Comment by blakeneyops** [comment]
This is happening on version 0.4.2

**Comment by mitchellh** [comment]
I'm pretty confident this is fixed in master (0.5, which will be released shortly). Do you mind testing with that?

**Comment by blakeneyops** [comment]
Sure, I will update once I confirm. Thanks.

**Comment by mitchellh** [comment]
Closing due to lack of response, but will reopen if there is one.

**Comment by blakeneyops** [comment]
Confirmed that this was resolved in 0.5.0. Thanks!
